python-package/lightgbm/basic.py

```diff
@@ -1383,8 +1439,15 @@ def _lazy_init(self, data, label=None, reference=None,
             self.__init_from_csc(data, params_str, ref_dataset)
         elif isinstance(data, np.ndarray):
             self.__init_from_np2d(data, params_str, ref_dataset)
-        elif isinstance(data, list) and len(data) > 0 and all(isinstance(x, np.ndarray) for x in data):
-            self.__init_from_list_np2d(data, params_str, ref_dataset)
+        elif isinstance(data, Sequence):
```

Comment:
TODO: revert `is_class` implementation 

This would force any 3rd party lib (presumably Data Access Layer libs) to import LightGBM.

Reply:
Generally we need an duck typing implementation.
 `typechecked` and `is_of_type` would do the job yet requires new dependency [pytypes](https://pypi.org/project/pytypes/).